client/my-sites/email/gsuite-add-users/index.jsx

```diff
@@ -21,30 +21,37 @@ import { fetchBySiteId } from 'state/google-apps-users/actions';
 import { getBySite, isLoaded } from 'state/google-apps-users/selectors';
 import { getDecoratedSiteDomains, isRequestingSiteDomains } from 'state/sites/domains/selectors';
 import { getSelectedSite } from 'state/ui/selectors';
-import { hasGSuiteSupportedDomain } from 'lib/domains/gsuite';
+import { hasGSuite, hasGSuiteSupportedDomain } from 'lib/domains/gsuite';
 import Main from 'components/main';
 import PageViewTracker from 'lib/analytics/page-view-tracker';
 import QuerySiteDomains from 'components/data/query-site-domains';
 import SectionHeader from 'components/section-header';
 
 class GSuiteAddUsers extends React.Component {
 	componentDidMount() {
-		const { domains, isRequestingDomains, selectedSite } = this.props;
-		this.redirectIfCannotAddEmail( domains, isRequestingDomains );
+		const { domains, isRequestingDomains, selectedDomainName, selectedSite } = this.props;
+		this.redirectIfCannotAddEmail( domains, isRequestingDomains, selectedDomainName );
 		this.props.fetchGoogleAppsUsers( selectedSite.ID );
 	}
 
 	shouldComponentUpdate( nextProps ) {
-		const { domains, isRequestingDomains } = nextProps;
-		this.redirectIfCannotAddEmail( domains, isRequestingDomains );
+		const { domains, isRequestingDomains, selectedDomainName } = nextProps;
+		this.redirectIfCannotAddEmail( domains, isRequestingDomains, selectedDomainName );
 		if ( isRequestingDomains || ! domains.length ) {
 			return false;
 		}
 		return true;
 	}
 
-	redirectIfCannotAddEmail( domains, isRequestingDomains ) {
-		if ( isRequestingDomains || hasGSuiteSupportedDomain( domains ) ) {
+	redirectIfCannotAddEmail( domains, isRequestingDomains, selectedDomainName ) {
+		let selectedDomainHasGSuite = false;
+		if ( selectedDomainName ) {
+			const selectedDomain = domains.reduce( function( selected, domain ) {
+				return domain.name === selectedDomainName ? domain.name : selected;
+			}, '' );
+			selectedDomainHasGSuite = hasGSuite( selectedDomain );
```

Comment:
This block of code looks a bit convoluted to me. I have the feeling we could use `array.find()` instead here:

```javascript
selectedDomainHasGSuite = domains.find( function( domain ) {
  if ( domain.name !== selectedDomainName ) {
    return false;
  }

  return hasGSuite( domain.name );
} );
```

Reply:
Didn't even think to use that because fo IE support.  I assume we polyfill or transpile that?  Looked it up and it is used in a couple places.  Good call.

Reply:
Didn't even think that this method might not be supported in all browsers 😅. I bet Babel transpile that though.